knitout-to-kcode.js

```diff
@@ -159,32 +155,33 @@ function merge_types(a,b) { //*
 	}
 
 	//types that only define one bed get merged:
+	//TODO: determing whether TYPE_SPLIT_TO_BACK can be merged with TYPE_x_KNIT, or only TYPE_KNIT_x (and vice versa)
 	if (a === TYPE_KNIT_x) {
 		if (b === TYPE_x_KNIT || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
 		else if (b === TYPE_x_TUCK || b === TYPE_KNIT_TUCK) return TYPE_KNIT_TUCK;
-		else if (b === TYPE_SPLIT_TO_BACK) return TYPE_KNITFRONT_SPLIT2BACK; //* //?
+		else if (b === TYPE_SPLIT_TO_BACK) return TYPE_SPLIT_TO_BACK;
 	} else if (a === TYPE_x_KNIT) {
 		if (b === TYPE_KNIT_x || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
 		else if (b === TYPE_TUCK_x || b === TYPE_TUCK_KNIT) return TYPE_TUCK_KNIT;
-		else if (b === TYPE_SPLIT_TO_FRONT) return TYPE_KNITBACK_SPLIT2FRONT; //* //?
+		else if (b === TYPE_SPLIT_TO_FRONT) return TYPE_SPLIT_TO_FRONT;
 	} else if (a === TYPE_TUCK_x) {
 		if      (b === TYPE_x_KNIT || b === TYPE_TUCK_KNIT) return TYPE_TUCK_KNIT;
 		else if (b === TYPE_x_TUCK || b === TYPE_TUCK_TUCK) return TYPE_TUCK_TUCK;
 	} else if (a === TYPE_x_TUCK) {
 		if      (b === TYPE_KNIT_x || b === TYPE_KNIT_TUCK) return TYPE_KNIT_TUCK;
 		else if (b === TYPE_TUCK_x || b === TYPE_TUCK_TUCK) return TYPE_TUCK_TUCK;
-	} else if (a === TYPE_KNIT_KNIT) { //* can split append to passes with knitting on both beds //?
+	} else if (a === TYPE_KNIT_KNIT) { //TODO: determine if split can be appended to passes with knitting on both beds
 		if (b === TYPE_KNIT_x || b === TYPE_x_KNIT) return TYPE_KNIT_KNIT;
 	} else if (a === TYPE_KNIT_TUCK) {
 		if (b === TYPE_KNIT_x || b === TYPE_x_TUCK) return TYPE_KNIT_TUCK;
 	} else if (a === TYPE_TUCK_KNIT) {
 		if (b === TYPE_TUCK_x || b === TYPE_x_KNIT) return TYPE_TUCK_KNIT;
 	} else if (a === TYPE_TUCK_TUCK) {
 		if (b === TYPE_TUCK_x || b === TYPE_x_TUCK) return TYPE_TUCK_TUCK;
-	} else if (a === TYPE_SPLIT_TO_BACK || a === TYPE_KNITFRONT_SPLIT2BACK) {
-		if (b === TYPE_SPLIT_TO_BACK || b === TYPE_KNITFRONT_SPLIT2BACK || b === TYPE_KNIT_x) return TYPE_KNITFRONT_SPLIT2BACK; //don't have to worry about a === TYPE_SPLIT_TO_BACK && b === TYPE_SPLIT_TO_BACK returning TYPE_KNITFRONT_SPLIT2BACK bc already taken care of by: if (a === b) return a;
-	} else if (a === TYPE_SPLIT_TO_FRONT || a === TYPE_KNITBACK_SPLIT2FRONT) {
-		if (b === TYPE_SPLIT_TO_FRONT || b === TYPE_KNITBACK_SPLIT2FRONT || b === TYPE_x_KNIT) return TYPE_KNITBACK_SPLIT2FRONT;
+	} else if (a === TYPE_SPLIT_TO_BACK || a === TYPE_SPLIT_TO_BACK) {
```

Comment:
This line (+ the next three) check a or b for the same type twice; e.g., `(a === TYPE_SPLIT_TO_BACK || a === TYPE_SPLIT_TO_BACK)` probably should edit these to just check once.

Reply:
oops, accidentally spaced on that when I was replacing `TYPE_KNITFRONT_SPLIT2BACK` with `TYPE_SPLIT_TO_BACK` etc.